### ISSUE TITLE
Go install doesn't work for telemetrygen

### ISSUE THREAD

**Issue by theletterf** [open-telemetry/opentelemetry-collector-contrib#27855]
### Component(s)

cmd/telemetrygen

## Description

I've tried the following command from the telemetrygen README file:

```
go install github.com/open-telemetry/opentelemetry-collector-contrib/cmd/telemetrygen@latest
```

On Ubuntu 23.10, with Go 1.21.1, this is what I get:

```
go: github.com/open-telemetry/opentelemetry-collector-contrib/cmd/telemetrygen@latest (in github.com/open-telemetry/opentelemetry-collector-contrib/cmd/telemetrygen@v0.87.0):
	The go.mod file for the module providing named packages contains one or
	more replace directives. It must not contain directives that would cause
	it to be interpreted differently than if it were the main module.
```

This makes it rather difficult if you only need that binary. I suspect this might happen to more packages. 

I ended up pulling the entire contrib repo and running `go build` from the telemetrygen directory. That gave the binary I needed for a get started tutorial.

### Collector version

0.87.0

## Environment
OS: Ubuntu 23.10
Compiler(if manually compiled): Go 1.21.1

**Comment by mx-psi** [comment]
This happens since #26668 because of https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/1842e1356889d1d35d007838f4f74edb2315d974/cmd/telemetrygen/main_test.go#L26

I think we can split these tests into a separate module. What do you think @codeboten @atoulme ?

**Comment by mx-psi** [comment]
#27878 did not fix this, but #28565 did. I intend to make a `v0.88.1` release only for telemetrygen so the install command works again

**Comment by mx-psi** [comment]
This is now fixed, I tagged v0.88.1: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/cmd/telemetrygen/v0.88.1
